### ISSUE TITLE
Multiple Accordions in the bootstrap4 have the identical html id "accordion"

### ISSUE THREAD

**Issue by xjlin0** [django-crispy-forms/django-crispy-forms#1092]
* Package version: 1.10.0
* Django version: 3.0
* Python version: 3.8
* Template pack: bootstrap4 (however bootstrap3 and bootstrap should be good)

### Description:

**Observation**: Multiple Accordions in the page have the identical html id "accordion", in bootstrap4 or bootstrap templates.
**Exceptions**: Multiple Accordions in the page should have the different html id
<img width="720" alt="Screen Shot 2020-12-12 at 9 13 01 AM" src="https://user-images.githubusercontent.com/8908273/101990324-33949c80-3c5b-11eb-8073-a3c85e9b78ca.png">

### Preferably also include:

- [x ] Example Django Crispy Forms code
- [x ] Screenshots
- [x ] Actual HTML generated
- [x ] Expected HTML

**Form generation**:
```python
self.helper.layout = Layout(
    Accordion(
        AccordionGroup(
            # this is for a relation 1......
        ), #css_id won't help
    ),
    Accordion(
        AccordionGroup(
            # this is for a relation 2......
        ), #css_id won't help
    ),
)
```

**produce**:
```html
<div id="accordion" role="tablist">
    <!-- this is for a relation 1...... -->
</div>
<div id="accordion" role="tablist">
    <!-- this is for a relation 2...... -->
</div>
```

Maybe it is hard code from:  /templates/bootstrap4/accordion.html
```html
<div id="accordion" role="tablist">
```
Maybe I can supply a PR later.

**Comment by cpina** [comment]
I haven't used "accordions" on django-crispy-forms. Yesterday I was investigating something else and I saw this code: https://github.com/django-crispy-forms/django-crispy-forms/blob/master/crispy_forms/bootstrap.py#L351 . I don't know if this approach might help for your case.

**Comment by xjlin0** [comment]
Hmmm... once I add `{{ accordion.css_id }}` in bootstrap4/accordion.html, the toggle of the accordion stop working.  It's because the JS look for according groups' parent by hard code id `#accordion`.  Therefore, the current situation is,  either my settings are off, or currently multiple accordions in a page is not supported.  :(
<img width="709" alt="Screen Shot 2020-12-12 at 11 20 45 AM" src="https://user-images.githubusercontent.com/8908273/101993169-6182dc80-3c6d-11eb-906b-140857fe490b.png">

**Comment by cpina** [comment]
I've just prepared an HTML with 4 accordions with the correct parent_data and they work fine. I didn't need to change any JavaScript.

Two accordions:
```html
<div id="accordion-01" role="tablist"> <div class="card mb-2"> <div class="card-header" role="tab"> <h5 class="mb-0"> <a data-toggle="collapse" href="#test-01" aria-expanded="true"
               aria-controls="test-01">
                test
            </a> </h5> </div> <div id="test-01" class="collapse show" role="tabpanel"
         aria-labelledby="test-01" data-parent="#accordion-01"> <div class="card-body">
            test
        </div> </div>
</div>

</div>
<div id="accordion-02" role="tablist"> <div class="card mb-2"> <div class="card-header" role="tab"> <h5 class="mb-0"> <a data-toggle="collapse" href="#test-02" aria-expanded="true"
               aria-controls="test-02">
                test2
            </a> </h5> </div> <div id="test-02" class="collapse show" role="tabpanel"
         aria-labelledby="test-02" data-parent="#accordion-02"> <div class="card-body">
            test2
        </div> </div>
</div>
```

In django-crispy-forms, and in order to have the `data-parent` we should modify `accordion-group.html` but this is not enough: `data-parent` is not in that context. I haven't properly investigated much more at the moment but the group would need to know the parent ID at that time for what I can see.

When I have time I might have a better look, so if anyone else wanted to fix it or send a suggestion please go ahead! :-)

**Comment by cpina** [comment]
This needs discussion but using https://github.com/cpina/django-crispy-forms/commit/628c75160aa1c6c612690188616dae3c8a76a930 fixes it for me.

It helped me to draw the hierarchy classes involved in this. Sorry for not doing ASCII art:

Accordion isA ContainerHolder isA Div (hasA css_id) isA LayoutObject

AccordionGroup isA Container isA Div (hasA css_id) isA LayoutObject

Code example:
```
        Accordion(
            AccordionGroup("test",
                AppendedText('appended_text', '.00'),
            )
        ),
```

What the commit is doing:
 * Accordion `__init__` sets the self.css_id to `f"accordion-{randint}"` (this was done before in `render`)
 * Accordion `__init__` sets parent_id to the AccordionGroup (built before)
 * Changed templates to use them

Moving the generation of randon int to the `__init__` has a consequence: before each rendering had a different accordion_id and it could not be known by code (e.g. generating JavaScript in the Python side after having it constructed and before having it rendered).

If someone did:
```
accordion = Accordion(
            AccordionGroup("test",
                AppendedText('appended_text', '.00'),
            )
        )
```

And used it twice in the same page: it wouldn't work (duplicated id). Not before (because the `AccordionGroup` didn't have the `data-parent`) and not now (because duplicated Ids). But I don't think that creating an Accordion object and using it twice in the same page is intended to work.

**Comment by smithdc1** [comment]
Fixed in #1099
